compiler/src/formatting/fmt.re

```diff
@@ -3992,10 +3992,36 @@ let print_program = (fmt, parsed_program) => {
     };
 
   group @@
-  fmt.print_comment_range(
-    fmt,
-    enclosing_start_location(parsed_program.prog_loc),
-    parsed_program.module_name.loc,
+  concat_map(
+    ~lead=
+      first =>
+        fmt.print_comment_range(
+          fmt,
+          enclosing_start_location(parsed_program.prog_loc),
+          first.attr_loc,
+        ),
+    ~sep=
+      (prev, next) =>
+        fmt.print_comment_range(
+          fmt,
+          ~none=hardline,
+          ~lead=space,
+          ~trail=hardline,
+          prev.Asttypes.attr_loc,
+          next.attr_loc,
+        ),
+    ~trail=
+      prev =>
+        fmt.print_comment_range(
+          fmt,
+          ~none=hardline,
+          ~lead=space,
+          ~trail=hardline,
```

Comment:
I'm pretty sure that the `~none` and/or `~trail` being hardline here are causing your extra newline in the attributes.

Reply:
Doesn't look like that fixed it; I think it's the fact that `print_comment_range` inserts `phatom_hardline ++ phantom_hardline` since `line_delta` > 1 between the comment and module header